### ISSUE TITLE
Screen is reversed by 180 degree on old Android devices [GLES3]

### ISSUE THREAD

**Issue by Solvero** [godotengine/godot#17083]
**Godot version:**
3.0.0 and 3.0.1

**OS/device including version:**
Manjaro Linux
Xperia T3 (D5103), Android 4.4.4

**Issue description:**
When game is exported to Android, displayed screen is rotated by 180 deg.
This affects only image. Touch screen seems to work correctly.
For example:
When I put a button at top-left corner of the screen it is visible at bottom-right corner.
To press the button I still have to touch top-left corner.

**Steps to reproduce:**
The following problem appeared to me when stable 3.0 was released.
It worked fine on alpha versions.

**Minimal reproduction project:**
[test.zip](https://github.com/godotengine/godot/files/1763988/test.zip)

**Comment by Solvero** [comment]
The following problem still exists on version 3.0.2 on Xperia T3.
I have performed test on different phone (Moto G5S Android 7.1.1) and it works fine there.

**Comment by ChrisWhatever1980** [comment]
I encountered the same problem with Godot 3.0.2 on a Samsung S4 with Android 4.4.2. 
Tried setting handheld orientation to reverse_landscape, which just rotated the problem 180 degrees.
Maybe caused by older android versions?
[TouchButtonTest.zip](https://github.com/godotengine/godot/files/1926475/TouchButtonTest.zip)

**Comment by ChrisWhatever1980** [comment]
I have helped myself with a workaround (see attached sample as well):
1. set orientation in project settings to reverse_landscape
2. Put TouchScreenButton into new scene and invert touch position in _input:

```func _input(event):
    if event is InputEventScreenTouch:
        var resolution = Vector2(
            ProjectSettings.get_setting("display/window/size/width"),
            ProjectSettings.get_setting("display/window/size/height"))
        #switch the following two lines to test on PC
        #var pos = event.get_position()
        var pos = Vector2(resolution.x - event.get_position().x, resolution.y - event.get_position().y)
        var ori = get_global_transform().get_origin()
        var dim = normal.get_size() * get_global_transform().get_scale()
        var rect = Rect2(ori, dim)
        if rect.has_point(pos):
            if event.pressed:
                emit_signal("ButtonPressed", self)
            else:
                emit_signal("ButtonReleased", self)
```

[TouchButtonWorkaround.zip](https://github.com/godotengine/godot/files/1926994/TouchButtonWorkaround.zip)

**Comment by Solvero** [comment]
Yes... I'm afraid that old Android causes problem. 
Nice workaround :smiley:. it solves button touchScreenButton.
However reverse_landscape screen causes that Android bar is at the bottom of the screen instead of top.

**Comment by MednauN** [comment]
Confirmed this problem on
Redmi Note 3 (Android 5.0.2)
Asus ZenPad 10 P01T (Android 5.0.1)

I tried to do some research the problem, here is what I found out:
1. The problem appears only on some devices.
2. Actually, the screen rotation here should be handled by the OS. And it's being handled, but somehow fails.
3. The layout and native controls are rotated just fine, the problem lies within `GLSurfaceView`.
4. When I tried to render something inside `GodotView` "by hand" using GLES 2.0 API, the rotation started to work for the entire scene. I assume, it may be related to poor GLES 3.0 support by these devices. Maybe, something is not refreshed correctly and draw call from GLES 2.0 API forces the refresh.

So I added the rendering of an invisible triangle inside `Renderer` class in `GodotView.java` and it did the trick for me. Another weird bug and even more weird fix. ¯\\_(ツ)_/¯

Here is my code, if someone wants to use it:
```java
//GodotView.java:663
private static class Renderer implements GLSurfaceView.Renderer {

	float[] verticesData = { 2.0f, 0.5f, 0.0f, -0.5f, 1.5f, 0.0f, 2.5f, -0.5f, 0.0f };
	FloatBuffer vertices = ByteBuffer.allocateDirect(verticesData.length * 4)
			.order(ByteOrder.nativeOrder()).asFloatBuffer().put(verticesData);

	public void onDrawFrame(GL10 gl) {
		vertices.position(0);
		GLES20.glVertexAttribPointer(0, 3, GLES20.GL_FLOAT, false, 0, vertices);
		GLES20.glEnableVertexAttribArray(0);
		GLES20.glDrawArrays(GLES20.GL_TRIANGLES, 0, 3);
		// The rest of onDrawFrame
```

**Comment by wrbeck** [comment]
I can confirm the issue: screen renders 180 degrees oposite to the touch controls on HTC M8.
Upshot is that performance is way smoother than on my 6 year old iMac. And export worked so easy and fast. Kudos to Godot; just need to fix this layout issue.

**Comment by robertiancu** [comment]
will be fixed this bug in next release?

**Comment by reduz** [comment]
Can you try using the GLES2 renderer?

**Comment by KiwoTez** [comment]
Using the GLES2 renderer option in the latest build solved the problem for me, tested on a LG G2.

**Comment by DDru** [comment]
Hello. I has same problem in Godot 3.0.6 and Samsung s3

**Comment by akien-mga** [comment]
It appears to be fixed with GLES2, so it's not critical for Godot 3.1. If someone can find a fix for those devices on GLES3 that would be great, but low priority given that the GLES3 renderer is going to be replaced by the Vulkan renderer.

**Comment by akien-mga** [comment]
As per the above, the recommended workaround for such devices would be to use the GLES2 renderer. As such I'll close this issue as the GLES3 renderer will be replaced in 4.0, so this issue is unlikely to be fixed (a PR is still welcome if anyone wants to fix it in GLES3 for a 3.2.x release).

**Comment by joined72** [comment]
@MednauN I need to fix this issue on 4.2.1 compatibility render ([take a look here](https://github.com/godotengine/godot/issues/86112)), can you help me explaining how you fixed it, please?
